components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -709,31 +752,101 @@ impl RegionInfoProvider for RegionInfoAccessor {
                 })
             })
     }
+
+    fn find_region_by_key(&self, key: &[u8]) -> Result<Option<Region>> {
+        let (tx, rx) = mpsc::channel();
+        let msg = RegionInfoQuery::GetRegionByKey {
```

Comment:
Why not use `RegionInfoQuery::SeekRegion`?

Reply:
SeekRegion finds all regions whose end key is larger than `key` where `key` may not be located in any of them. It's not convenient to use it in this case.

Reply:
Isn't it just
```
self.seek_region(key, Box::new(|iter| {
   if let Some(info) = iter.next() && info.region.get_start_key() < key {
        tx.send(info.region.clone());
    }
}))
```

Reply:
But, after encapsulating it in `RegionInfoProvider`, it just needs one line.

Reply:
So you use 23 lines + 2 lines + 21 lines to encapsulate 4 lines?

Reply:
find_region_by_key is a common pattern that may be used in many places.

Reply:
If it's a common pattern, then it won't be added 3 years after `SeekRegion` is introduced.